### PR TITLE
Add new layer (auto-insert-template)

### PR DESCRIPTION
**Pull request by YasuharuIida**
- This layer provides template auto-insertion.(template file is similar to yasnippet's snippet file)

### CODE REVIEW

File: layers/+completion/auto-insert-template/packages.el [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defconst auto-insert-template-packages
+  '(
+    (autoinsert :location built-in)
+    yasnippet
```
Comment (TheBB) [inline layers/+completion/auto-insert-template/packages.el]: If there's no function associated with these packages in your packages.el file, they shouldn't be listed in the package list either.

File: layers/+completion/auto-insert-template/config.el [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Yasuharu Iida <YasuharuIida@users.noreply.github.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(setq yatemplate-dir (concat configuration-layer-private-directory "templates"))
```
Comment (TheBB) [inline layers/+completion/auto-insert-template/config.el]: `defvar`, not `setq`

File: layers/+completion/auto-insert-template/README.org [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#+TITLE: auto-insert-template layer
+#+HTML_HEAD_EXTRA: <link rel="stylesheet" type="text/css" href="../css/readtheorg.css" />
```
Comment (TheBB) [inline layers/+completion/auto-insert-template/README.org]: This is auto generated on export, no need to have it here.

File: layers/+completion/auto-insert-template/README.org [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+* Table of Contents                                        :TOC_4_org:noexport:
```
Comment (TheBB) [inline layers/+completion/auto-insert-template/README.org]: Use `TOC_4_gh`.

**Comment by TheBB** [comment]
Looks all right except my comments. :+1:

**Comment by YasuharuIida** [comment]
Yatemplate depends on `autoinsert` and `yasnippet` package.

**Comment by YasuharuIida** [comment]
* Fix
- `setq` to `defvar`
- remove HTML haeder

**Comment by YasuharuIida** [comment]
Whether new layer or part of auto-completion layer?

**Comment by YasuharuIida** [comment]
OK! I will add this layer's feature to auto-completion layer.

**Comment by TheBB** [comment]
Well I just meant it could be discussed, I wasn't really suggesting one or the other. But okay. :-)

**Comment by TheBB** [comment]
> Yatemplate depends on `autoinsert` and `yasnippet` package.

You still don't need to list them unless you specifically need to configure those packages.

**Comment by YasuharuIida** [comment]
Sorry. I misunderstood.

**Pull request closed without merging**